libraries/Microsoft.Bot.Builder.Ai.QnA/QnAMakerTraceInfo.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Ai.QnA
+{
+
+    /// <summary>
+    /// This class represents all the trace info that we collect from the LUIS Recognizer Middleware
+    /// </summary>
+    public class QnAMakerTraceInfo
```

Comment:
Please add a serialization test around this. Compare the output serilization with a hand-rolled JSON string and make sure it's right. This will prevent future regressions.

Reply:
Done.